src/pkg/validation/validation_command.go

```diff
@@ -36,27 +41,30 @@ func ValidationCommand(inputFile string) (validationResponse ValidationResponse,
 	}
 	validationResponse.Validator = validator
 
-	// Get and set version warnings
-	version := validator.GetSchemaVersion()
-	err = versioning.VersionWarning(version)
-	if err != nil {
-		validationResponse.Warnings = append(validationResponse.Warnings, err.Error())
-	}
-
 	// Set the document path
 	validator.SetDocumentPath(inputFile)
 
 	// Run the validation
-	validationError := validator.Validate()
-
-	// Write validation result if it was specified and exists before returning ValidateCommand error
-	validationResult, _ := validator.GetValidationResult()
-	validationResponse.Result = validationResult
+	err = validator.Validate()
+	if err != nil {
+		validationError, ok := err.(*jsonschema.ValidationError)
+		// If the error is not a validation error, return the error
+		if !ok {
+			return validationResponse, err
+		}
+		// Set the jsonschema error in the validation response
+		validationResponse.JsonSchemaError = validationError
+	}
 
-	// Handle the validation error
-	if validationError != nil {
-		return validationResponse, fmt.Errorf("failed to validate %s version %s: %s", validator.GetModelType(), validator.GetSchemaVersion(), err)
+	// Get and set version warnings if upgrade available
+	version := validator.GetSchemaVersion()
+	err = versioning.VersionWarning(version)
+	if err != nil {
+		validationResponse.Warnings = append(validationResponse.Warnings, err.Error())
 	}
 
+	// Set the response result, ignore error since we know validate has been run
+	validationResponse.Result, _ = validator.GetValidationResult()
```

Comment:
Feel like this may be safer long term:
```golang

	// Get the validation result
	validationResponse.Result, err = validator.GetValidationResult()
	// If there is an error, return it, but there shouldn't be an error
	if err != nil {
		return validationResponse, fmt.Errorf("shouldn't error,check the GetValidationResult method for more information: %s", err)
	}
```

Reply:
created an issue to look at addressing the root of this issue #268.